src/lib/utils/assert.ts

```diff
@@ -0,0 +1,11 @@
+/**
+ * Asserts that `condition` is truthy.
+ * @param condition The condition to assert.
+ * @param message Error message to throw on assertion error.
+ * @throw An error if `condition` is falsy.
+ */
+export default function assert(condition: unknown, message = 'Assertion Error'): asserts condition {
```

Comment:
+1 for assertions! 🤘I might consider renaming in case we'll need to add more in the future (assertNotNull, assertNotNullOrEmpty, etc.), but we can always do it later.